aviatrix/resource_aviatrix_spoke_external_device_conn.go

```diff
@@ -1113,6 +1235,74 @@ func resourceAviatrixSpokeExternalDeviceConnUpdate(d *schema.ResourceData, meta
 		}
 	}
 
+	enableBfd, ok := d.Get("enable_bfd").(bool)
+	if !ok {
+		return fmt.Errorf("expected enable_bfd to be a boolean, but got %T", d.Get("enable_bfd"))
+	}
+	if enableBfd && connType != "bgp" {
+		return fmt.Errorf("BFD is only supported for BGP connection type")
+	}
+	if enableBfd {
+		// get the new BGP BFD config
+		bgpBfdConfig, ok := d.Get("bgp_bfd").([]interface{})
+		if !ok {
+			return fmt.Errorf("expected bgp_bfd to be a list of maps, but got %T", d.Get("bgp_bfd"))
+		}
+		var bgpBfdConfigList []*goaviatrix.BgpBfdConfig
+		// Update the BGP BFD config if bfd is enabled and config has changed
+		if len(bgpBfdConfig) > 0 && d.HasChange("bgp_bfd") {
+			for _, v := range bgpBfdConfig {
+				bfdConfig := v.(map[string]interface{})
+				transmitInterval, ok := bfdConfig["transmit_interval"].(int)
+				if !ok {
+					transmitInterval = defaultBfdTransmitInterval
+				}
+				receiveInterval, ok := bfdConfig["receive_interval"].(int)
+				if !ok {
+					receiveInterval = defaultBfdReceiveInterval
+				}
+				multiplier, ok := bfdConfig["multiplier"].(int)
+				if !ok {
+					multiplier = defaultBfdMultiplier
+				}
+				bgpBfdConfigList = append(bgpBfdConfigList, &goaviatrix.BgpBfdConfig{
+					TransmitInterval: transmitInterval,
+					ReceiveInterval:  receiveInterval,
+					Multiplier:       multiplier,
+				})
+			}
+		} else {
+			// set the bgp bfd config using the default values
+			bgpBfdConfigList = append(bgpBfdConfigList, &goaviatrix.BgpBfdConfig{
+				TransmitInterval: defaultBfdTransmitInterval,
+				ReceiveInterval:  defaultBfdReceiveInterval,
+				Multiplier:       defaultBfdMultiplier,
+			})
+		}
+		externalDeviceConn := &goaviatrix.ExternalDeviceConn{
+			GwName:         d.Get("gw_name").(string),
+			ConnectionName: d.Get("connection_name").(string),
+			EnableBfd:      d.Get("enable_bfd").(bool),
+			BgpBfdConfig:   bgpBfdConfigList,
+		}
+		err := client.EditConnectionBgpBfd(externalDeviceConn)
+		if err != nil {
+			return fmt.Errorf("could not update BGP BFD config: %v", err)
+		}
+	} else {
+		if d.HasChange("enable_bfd") {
```

Comment:
Shouldn't we always force the terraform desired state onto the controller? If the BFD was enabled out of band but not in terraform, then would a terraform apply set it back?

Reply:
I have tweaked the bgp_bfd config. Open to suggestions on how I can improve it further.

Reply:
Updated the code to consider bgp_bfd changes